iroha/src/torii.rs

```diff
@@ -224,43 +289,55 @@ async fn handle_health(
 async fn handle_pending_transactions_on_leader(
     state: State<ToriiState>,
     _path_params: PathParams,
-    _query_params: QueryParams,
+    query_params: QueryParams,
     _request: HttpRequest,
 ) -> Result<HttpResponse, String> {
-    if state.read().await.sumeragi.read().await.is_leader() {
-        Ok(HttpResponse::ok(
-            Headers::new(),
+    let range = match Pagination::try_from(&query_params) {
+        Ok(range) => range,
+        Err(e) => {
+            log::error!("Failed to decode pagination: {}", e);
+            // Is it really internal server error?
+            return Ok(HttpResponse::internal_server_error());
+        }
+    };
+
+    let PendingTransactions(pending_transactions) =
+        if state.read().await.sumeragi.read().await.is_leader() {
             state
                 .read()
                 .await
                 .transactions_queue
                 .read()
                 .await
                 .pending_transactions()
-                .into(),
-        ))
-    } else {
-        let pending_transactions: PendingTransactions = Network::send_request_to(
-            state
-                .read()
-                .await
-                .sumeragi
-                .read()
-                .await
-                .network_topology
-                .leader()
-                .address
-                .as_ref(),
-            Request::new(uri::PENDING_TRANSACTIONS_URI.to_string(), Vec::new()),
-        )
-        .await?
-        .into_result()?
-        .try_into()?;
-        Ok(HttpResponse::ok(
-            Headers::new(),
-            pending_transactions.into(),
-        ))
-    }
+        } else {
+            Network::send_request_to(
+                state
+                    .read()
+                    .await
+                    .sumeragi
+                    .read()
+                    .await
+                    .network_topology
+                    .leader()
+                    .address
+                    .as_ref(),
+                Request::new(uri::PENDING_TRANSACTIONS_URI.to_string(), Vec::new()),
+            )
+            .await?
+            .into_result()?
+            .try_into()?
+        };
+
+    Ok(HttpResponse::ok(
+        Headers::new(),
+        PendingTransactions(match range {
+            Pagination::From(range) => pending_transactions[range].to_vec(),
+            Pagination::FromTo(range) => pending_transactions[range].to_vec(),
```

Comment:
Pending transactions should also be ordered.

Reply:
I guess queue is ordered by time by default. Should it be ordered by ids?